### ISSUE TITLE
s-lang missing dependency: pcre

### ISSUE THREAD

**Issue by a-m-s** [Linuxbrew/homebrew-core#11338]
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single formula? If it's a general `brew` problem please file this issue at Linuxbrew/brew: https://github.com/Linuxbrew/brew/issues/new/choose. If it's a tap (e.g. Brewsci/homebrew-bio) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)

Install byobu to a non-standard prefix. byobu needs to install s-lang.

- What happened (include command output)

s-lang fails to build:

```
/tmp/s-lang-20190123-87701-1uszggs/slang-2.3.1a/modules/pcre-module.c:26:18: fatal error: pcre.h: No such file or directory                                                                                                                 
 #include <pcre.h>
```

I did `brew install pcre` (after the other troubleshooting steps), and then s-lang built and installed correctly.

- What you expected to happen

Install s-lang dependencies before attempting the build.

- Step-by-step reproduction instructions (by running `brew install` commands)

Run `brew install s-lang` when pcre is not installed.

**Comment by iMichka** [comment]
Hi. Would you like to submit a pull request to add the dependency to the formula?

**Comment by a-m-s** [comment]
On further testing, this *seems* to be only a problem on machines where brew must first install gcc etc.

Is there some way to express that in the formula?

**Comment by iMichka** [comment]
Not sure what you mean? You can `depends_on "gcc"` if you need to, but you were talking about `pcre` here. Adding `depends_on "pcre"` did not help?

**Comment by a-m-s** [comment]
Adding the pcre dependency works great, for me, but it doesn't seem to be necessary for people who can use the bottle without building.

On the ancient Ubuntu 14.04 system (for which I do not have root) brew insists that it must first install gcc and whatnot, and then falls over pcre.

If I try to reproduce the problem on an Ubuntu 18.11 system then it does not require pcre to install, and even built successfully when I forced it (which presumably uses the host toolchain).

The question is: how do I form the dependency such that it fixes the problem on one system without adding pointless clutter on the other system?

**Comment by iMichka** [comment]
Looking at this, pcre is a dependency: http://www.linuxfromscratch.org/blfs/view/8.3/general/slang.html

What happens here is probably the following: when we build the bottle, we do that in an almost empty docker container, where there is no pcre. A-Slang probably does not complain about pcre not being there at that point.

If you happen to have pcre installed on the system, through another package maintainer, it will pick it up. Our superenv will then prevent linking against that pcre because it is outside the homebrew folder. The only way out is to explicitely depend on brewed pcre. Or check if there is a —without-pcre option somewhere.

It would have helped if you had provided the brew gist-logs of the failed build: we would have been able to detect that issue.

Adding the dependency to the formula is fine in that case.

**Comment by a-m-s** [comment]
Ew, configure leakage. Not nice, but makes sense. :-(

I guess the question then is whether we want it to support pcre, or not? I'm not sure what s-lang even does - I installed it as a dependency of something else. It does appear to have a `--without-pcre` option.

I didn't have the log any more by the time I'd manually fixed the problem. I can probably recreate it if it's still useful, but I'm short of space on the affected system.

**Comment by iMichka** [comment]
Fixed by #11463.
Let's got for the pcre dependency. We try to follow the linuxfromscratch instructions when possible.
